### PR TITLE
Update API

### PR DESCRIPTION
**Pull request by AGulev**
* Update API to Nakama 2.14.0
* Fix issue https://github.com/heroiclabs/nakama-defold/issues/14

### CODE REVIEW

File: nakama/nakama.lua [hunk 1]
```diff
@@ -5,6 +5,9 @@ local log = require "nakama.util.log"
 local async = require "nakama.util.async"
 local api_session = require "nakama.session"
 
+local uri = require "nakama.util.uri"
+local uri_encode = uri.encode
+
```
Comment (AGulev) [inline nakama/nakama.lua:10]: We use `uri_encode` in some functions
Issue https://github.com/heroiclabs/nakama-defold/issues/14

File: codegen/main.go [hunk 2]
```diff
@@ -32,6 +32,9 @@ local log = require "nakama.util.log"
 local async = require "nakama.util.async"
 local api_session = require "nakama.session"
 
+local uri = require "nakama.util.uri"
+local uri_encode = uri.encode
```
Comment (AGulev) [inline codegen/main.go:36]: We use `uri_encode` in some functions
Issue https://github.com/heroiclabs/nakama-defold/issues/14

File: codegen/main.go [hunk 3]
```diff
@@ -269,15 +278,15 @@ function M.{{ $operation.OperationId | pascalToSnake }}(
     {{- range $parameter := $operation.Parameters }}
 	{{- $varName := varName $parameter.Name $parameter.Type $parameter.Schema.Ref }}
     {{- if eq $parameter.In "path" }}
-	url_path = url_path.gsub("{{- print "{" $parameter.Name "}"}}", uri_encode({{ $varName }}))
+	url_path = url_path:gsub("{{- print "{" $parameter.Name "}"}}", uri_encode({{ $varName | pascalToSnake }}))
```
Comment (AGulev) [inline codegen/main.go:281]: `string.gsub (s, pattern, repl [, n])`
`string:gsub (pattern, repl [, n])`

File: codegen/main.go [hunk 4]
```diff
@@ -344,6 +353,11 @@ func pascalToSnake(input string) (output string) {
 	return
 }
 
+func removePrefix(input string) (output string) {
+	output = strings.Replace(input, "nakama_", "", -1)
+	return
+}
```
Comment (AGulev) [inline codegen/main.go:359]: All operator Ids now has `Nakama_` prefix, but our module name `nakama` already
image

https://github.com/heroiclabs/nakama/commit/0ff7d3581c1ed90c7372e9f399bbd27462d34821#

File: codegen/main.go [hunk 5]
```diff
@@ -43,21 +46,25 @@ local M = {}
 function M.create_{{ $classname | pascalToSnake }}(
 	{{- range $propname, $property := $definition.Properties }}
 	{{- $luaType := luaType $property.Type }}
-	{{- $varName := varName $propname $property.Type $property.Ref }}
-	{{ $varName }}, -- '{{ $luaType }}' ({{ $property.Ref | cleanRef | pascalToSnake }}) {{ $property.Description }}
+	{{- $varName := varName  $propname $property.Type $property.Ref }}
+	{{- $varName := $varName | pascalToSnake }}
```
Comment (AGulev) [inline codegen/main.go:50]: Now we should use `pascalToSnake` a bit more often:
image
https://github.com/heroiclabs/nakama/commit/0ff7d3581c1ed90c7372e9f399bbd27462d34821#

**Comment by novabyte** [comment]
This looks really good @AGulev. Thanks for the contribution! 🙏 

Can you add a CHANGELOG entry for the updates and would you like to start a contributor file or add yourself to the README as a contributor?

**Comment by AGulev** [comment]
> Can you add a CHANGELOG entry for the updates and would you like to start a contributor file or add yourself to the README as a contributor?

I can do it ofc., but with Defold libraries/extentsions, we usually use [releases](https://github.com/heroiclabs/nakama-defold/releases) page to track changes. Maybe it would be better to be consistent with the other Defold libs?

**Comment by novabyte** [comment]
Yep we also use releases as a typical distribution workflow in our other client SDKs:

https://github.com/heroiclabs/nakama-dotnet/releases

And we maintain a CHANGELOG for the SDK as well:

https://github.com/heroiclabs/nakama-dotnet/blob/master/CHANGELOG.md

I will standardize the workflow with this SDK to match the others we maintain moving forwards.

**Comment by AGulev** [comment]
Ok! Done! Thanks!

**Comment by britzl** [comment]
Yes, the changes look good! Thanks Alexey!

**Comment by novabyte** [comment]
Thanks again @AGulev 🙏

**Pull request merged**